elastic/kibana
# [Logs onboarding] Install system integration when onboarding system logs - React query approach

Closes https://github.com/elastic/kibana/issues/163244.

### Changes
- Use `react-query` to fetch data from fleet apis, couldn't use the `useFetcher` hook because it's tailoring internal routes in the plugin. In this matter two hooks have been created: `useSystemIntegrationStatus` is responsible for checking the status of the integration returning `status` (whether is installed or not) and `version`; `useSystemIntegrationInstall` is responsible for installing the system integration.
- `system-integration-banner` component have been created to hold all the logic related to system integration.

#### After this changes

##### When integration is not installed

https://github.com/elastic/kibana/assets/1313018/4162319f-35d3-42d3-bd4d-821d1da26a8b

##### When integration is already installed

https://github.com/elastic/kibana/assets/1313018/5f1bf76e-7ed4-4f2c-ba4c-a8b2f3ff80a2

### How to test?

- Enter the [custom deployment](https://yngrdyn-deploy-kiban-pr163410.kb.us-west2.gcp.elastic-cloud.com)
- Check [installed integrations](https://yngrdyn-deploy-kiban-pr163410.kb.us-west2.gcp.elastic-cloud.com/app/integrations/installed) `/app/integrations/installed`
- Go to [observability onboarding](https://yngrdyn-deploy-kiban-pr163410.kb.us-west2.gcp.elastic-cloud.com/app/observabilityOnboarding) landing page: `app/observabilityOnboarding`
- Select `Stream host system logs` or `Quickstart`
- System integration should install (if it's not installed in the deployment) or just notify that has been installed (if it's already installed in the deployment)
- After entering `Stream host system logs` page System integration should be installed
